Server/utility.ts

```diff
@@ -0,0 +1,52 @@
+// Merge functions.js once it becomes functions.ts
+import { Coordinate, Events } from "./types.js";
+
+export function compArr<T>(c1: T[], c2: T[]) {
+    return c1.length === c2.length && c1.every((e, i)=>e === c2[i]);
+}
+
+// Modifies e1 in place by appending events from e2 to e1
+export function concatEvents(e1: Events, e2: Events) {
+    doubleIt((i, j)=>e1[i][j].push(...e2[i][j]), 0, 0, 2, 2);
+}
+
+// Makes a deep copy of objects and primitives
+// Does not work witih functions and symbols
+export function deepCopy<T>(obj: T): T {
+    if (typeof(obj === "object")) {
```

Comment:
I think you meant to write this:
```suggestion
    if (typeof obj === "object") {
```